src/internal/connector/onedrive/permission.go

```diff
@@ -129,11 +129,16 @@ func computeParentPermissions(
 	}
 }
 
+type postDeleteItemPermissioner interface {
```

Comment:
nit: Interface name is slightly misleading since it both deletes and updates item permisssions. How about `postItemPermissionUpdater`?

Also, on a side note, I feel like we should decouple update from delete in `UpdatePermissions`. Otherwise we will have niche interfaces such as this one. Maybe it's just that I am new to go, but it's a significant cognitive load keeping track of multiple interface names which serve a niche purpose.

Reply:
mmm, have to disagree.  It's a bit clunky as a name, but this interface composition is ultimately the same thing you'll see many places elsewhere in golang code.  Standard example, the io package: https://pkg.go.dev/io#ReadWriter.  Though, you're right, it doesn't have a post method.  Probably should be named `updateDeleteItemPermissioner`.  Or maybe `itemPermissionUpdaterdeleter`.

> Otherwise we will have niche interfaces such as this one.

Niche interfaces, especially those that are compositions of other more standardized interfaces, are an ideal, not a smell.  If a function needs to both delete and upload data in order to complete its process, and a single client is expected to supply both actions, then the shape of the interface required for that func is specifically those two methods: no more and no less.

Reply:
`updateDeleteItemPermissioner` sounds good